src/libpmemobj/memops.c

```diff
@@ -398,6 +398,20 @@ operation_add_buffer(struct operation_context *ctx,
 	size_t curr_size = MIN(real_size, ctx->ulog_curr_capacity);
 	size_t data_size = curr_size - sizeof(struct ulog_entry_buf);
 
+	size_t next_entry_offset = ctx->ulog_curr_offset +
+		ALIGN_UP(curr_size, CACHELINE_SIZE);
+	struct ulog_entry_base *next_entry = NULL;
+	if (next_entry_offset >= ctx->ulog_curr_capacity) {
```

Comment:
Is it possible that both of the below conditions are true?
next_entry_offset < ctx->ulog_curr_capacity
next_entry_offset + sizeof(struct ulog) >= ctx->ulog_curr_capacity

Reply:
Yes, if there's exactly one cacheline (so one ulog entry) remaining at the end of the log.